libs/core/kiln_ai/adapters/docker_model_runner_tools.py

```diff
@@ -0,0 +1,119 @@
+    """
+    try:
+        base_url = docker_model_runner_base_url()
+        # Docker Model Runner uses OpenAI-compatible endpoints
+        async with httpx.AsyncClient() as client:
+            response = await client.get(f"{base_url}/v1/models", timeout=5.0)
+            response.raise_for_status()
+    except httpx.RequestError:
+        return False
+    return True
+
+
+class DockerModelRunnerConnection(BaseModel):
+    message: str
+    version: str | None = None
+    supported_models: List[str]
+    untested_models: List[str] = Field(default_factory=list)
+
+    def all_models(self) -> List[str]:
+        return self.supported_models + self.untested_models
+
+
+# Parse the Docker Model Runner /v1/models response
+def parse_docker_model_runner_models(
+    models_response: Any,
+) -> DockerModelRunnerConnection | None:
+    # Build a list of models we support for Docker Model Runner from the built-in model list
+    supported_docker_models = [
+        provider.model_id
+        for model in built_in_models
+        for provider in model.providers
+        if provider.name == ModelProviderName.docker_model_runner
+    ]
+    # Note: Docker Model Runner aliases will be added when we configure models
+
+    if "data" in models_response:
+        models = models_response["data"]
+        if isinstance(models, list):
+            model_names = [model["id"] for model in models if "id" in model]
+            available_supported_models = []
+            untested_models = []
+
+            for model in model_names:
+                if model in supported_docker_models:
+                    available_supported_models.append(model)
+                else:
+                    untested_models.append(model)
+
+            if available_supported_models or untested_models:
+                return DockerModelRunnerConnection(
+                    message="Docker Model Runner connected",
+                    supported_models=available_supported_models,
+                    untested_models=untested_models,
+                )
+
+    return DockerModelRunnerConnection(
+        message="Docker Model Runner is running, but no supported models are available. Ensure models like 'ai/llama3.2:3B-Q4_K_M', 'ai/qwen3:8B-Q4_K_M', or 'ai/gemma3n:4B-Q4_K_M' are loaded.",
+        supported_models=[],
+        untested_models=[],
+    )
+
+
+async def get_docker_model_runner_connection() -> DockerModelRunnerConnection | None:
+    """
+    Gets the connection status for Docker Model Runner.
+    """
+    try:
+        base_url = docker_model_runner_base_url()
+        # Use OpenAI client to get models list
+        client = openai.OpenAI(
+            api_key="dummy",  # Docker Model Runner doesn't require API key
+            base_url=f"{base_url}/v1",
+            max_retries=0,
+        )
+        models_response = client.models.list()
+        # Convert to dict format for parsing
+        models_dict = {"data": [{"id": model.id} for model in models_response]}
```

Comment:
It's strange to go from a nice typed object into a dict, then to have a bunch of type safety checks in parse_docker_model_runner_models.

You can make parse_docker_model_runner_models take a `List[Model]` instead of Any and it will be a lot cleaner.